testgrid/cmd/transfigure/main.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright © 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"fmt"
+	"os"
+
+	transfigure "k8s.io/test-infra/testgrid/cmd/transfigure/cmd"
+)
+
+func main() {
+	if err := transfigure.Execute(); err != nil {
```

Comment:
Having a "main.go" and a "transfigure.go" file is a good idea, but I would change what goes into which file.

main.go would be a good place to handle flags and options. Summarizer has an [example](https://github.com/GoogleCloudPlatform/testgrid/blob/1844fdad5f0b8ddc2101e52069cca0209403e765/cmd/summarizer/main.go#L54) of this pattern.

Handling flags is complex, since users might specify these options with flags or in the old "list everything in the right order" way. Using Cobra for this seems great if that's what you're comfortable with, but the options parsing should pass the data it cares about (or an options struct) in the functions. Without that, everything is too coupled together to test easily.

Reply:
Done